packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -144,13 +150,22 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
         }
     }
 
-    protected async newTerminal(): Promise<void> {
-        const widget = <TerminalWidget>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
-            created: new Date().toString()
+    public async newTerminal(options: TerminalWidgetOptions): Promise<TerminalWidget & Widget> {
+        const widget = <TerminalWidgetImpl>await this.widgetManager.getOrCreateWidget(TERMINAL_WIDGET_FACTORY_ID, <TerminalWidgetFactoryOptions>{
+            created: new Date().toString(),
+            ...options
         });
+        return widget;
+    }
+
+    public activateWidget(widget: TerminalWidget & Widget): void {
         this.shell.addWidget(widget, { area: 'bottom' });
```

Comment:
What if the widget is already attached to another side-bar? It would be good to check it:
```ts
const tabBar = shell.getTabBarFor(widget);
if (!tabBar) {
    this.shell.addWidget(widget, { area: 'bottom' });
}
```